pkg/services/ngalert/notifier/channels/slack.go

```diff
@@ -339,3 +400,59 @@ func (sn *SlackNotifier) buildSlackMessage(ctx context.Context, as []*types.Aler
 func (sn *SlackNotifier) SendResolved() bool {
 	return !sn.GetDisableResolveMessage()
 }
+
+func (sn *SlackNotifier) slackFileUpload(ctx context.Context, data io.Reader, recipient, token string) error {
+	sn.log.Info("Uploading to slack via file.upload API")
+	headers, uploadBody, err := sn.generateSlackBody(data, token, recipient)
+	if err != nil {
+		return err
+	}
+	cmd := &models.SendWebhookSync{
+		Url: "https://slack.com/api/files.upload", Body: uploadBody.String(), HttpHeader: headers, HttpMethod: "POST",
```

Comment:
Could we extract this URL and make it a config option like we did with `SlackAPIEndpoint`?

The reason is we added support for alternative slack servers (like `slack.gov`) which breaks if this is hardcoded to `slack.com`. (Really wish we didn't extract out the whole operation URL into `SlackAPIEndpoint` because that now makes this a little awkward with the second endpoint)

Reply:
yeah, sgtm.

Reply:
I don't have time to add an UX that allows us to change this value in the UI. That needs to be another issue: https://github.com/grafana/grafana/issues/49376